charts/node/templates/statefulset.yaml

```diff
@@ -17,10 +17,31 @@ spec:
   serviceName: {{ $fullname }}
   template:
     metadata:
-      {{- with .Values.podAnnotations }}
+      {{- if or .Values.podAnnotations .Values.node.vault.keys .Values.node.vault.nodeKey }}
       annotations:
+      {{- with .Values.podAnnotations }}
       {{- toYaml . | nindent 8 }}
       {{- end }}
+      {{- range $keys := .Values.node.vault.keys }}
+        vault.hashicorp.com/agent-inject-secret-{{ .name }}: {{ .vaultPath | squote }}
+        vault.hashicorp.com/agent-inject-template-{{ .name }}: |
+          {{`{{ with secret "`}}{{ .vaultPath }}{{`" }}{{- range $k, $v := .Data.data }}
+          {{ $k }}: {{ $v }}
+          {{- end }}{{ end }}`}}
+      {{- end }}
+      {{- if or .Values.node.vault.keys .Values.node.vault.nodeKey }}
+      {{- if .Values.node.vault.nodeKey }}
+        vault.hashicorp.com/agent-inject-secret-{{ .Values.node.vault.nodeKey.name }}: {{ .Values.node.vault.nodeKey.vaultPath | squote }}
+        vault.hashicorp.com/agent-inject-template-{{ .Values.node.vault.nodeKey.name }}: |
+          {{`{{ with secret "`}}{{ .Values.node.vault.nodeKey.vaultPath }}{{`" }}{{- range $k, $v := .Data.data }}
```

Comment:
I'm not a big fan of this approach to pull all KV into a big pseudo YAML file. I think it might cause issue if the value contains spaces such as when using a seed phrase. I would prefer to save each secret that we use into its own file. I will push a PR with implementation of this method.